node/__tests__/request_coordinator.test.ts

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable node/no-unpublished-import */
+
+const originalDns = jest.requireActual('dns');
+
+const markedResolver = {
+  resolveSrv: jest.fn(() => Promise.resolve([{name: 'host', port: 0}])),
+  resolve4: jest.fn(() => Promise.resolve(['ip1', 'ip2'])),
+};
+
+jest.mock('dns', () => ({
+  ...originalDns,
+  promises: {
+    Resolver: jest.fn(() => markedResolver),
+  },
+}));
+
+const markedAdminClient = {
+  pause: jest.fn(() => Promise.resolve()),
+  unpause: jest.fn(() => Promise.resolve()),
+  checkPaused: jest.fn(() => Promise.resolve(false)),
+  getHost: jest.fn(() => 'host'),
+};
+
+jest.mock('../src/admin_client', () => ({
+  AdminClient: jest.fn(() => markedAdminClient),
+}));
+
+import {RequestCoordinator} from '../src/request_coordinator';
+
+beforeEach(() => {
+  markedAdminClient.pause.mockRestore();
+  markedAdminClient.unpause.mockRestore();
+});
+
+test('pause successfully', async () => {
+  const coordinator = new RequestCoordinator('srv');
+
+  await coordinator.pause(false);
+
+  expect(markedAdminClient.pause).toBeCalledWith(false, null);
+  expect(markedAdminClient.pause).toBeCalledTimes(2);
+});
+
+test('pause unsuccessfully', async () => {
+  markedAdminClient.pause = jest.fn(() => Promise.reject(new Error()));
+
+  const coordinator = new RequestCoordinator('srv');
+
+  await coordinator.pause(false);
+
+  expect(markedAdminClient.pause).toBeCalledWith(false, null);
+  expect(markedAdminClient.unpause).toBeCalledTimes(2);
+});
+
+test('unpause successfully', async () => {
+  const coordinator = new RequestCoordinator('srv');
+
+  await coordinator.unpause();
+
+  expect(markedAdminClient.unpause).toBeCalledTimes(2);
+});
+
+test('unpause unsuccessfully', async () => {
+  markedAdminClient.unpause = jest.fn(() =>
+    Promise.reject(new Error('unpause error'))
+  );
+
+  const coordinator = new RequestCoordinator('srv');
+
+  await expect(coordinator.unpause()).rejects.toThrow('unpause error');
+});
+
+test('checkPaused successfully', async () => {
+  const coordinator = new RequestCoordinator('srv');
+
+  const paused = await coordinator.checkPaused();
+
+  expect(paused).toEqual([
+    {host: 'host', paused: false},
```

Comment:
Can it be like ip1 resovles host1 and ip2 resolves host2 to verify they are properly resolved?

Reply:
Yes. Of course.
I revised it in https://github.com/scalar-labs/scalar-admin/pull/23/commits/f11e1efc6392942b86b2cf3dd7c09c9a09dfdbf0
Please take a look!